catalogs/catalog-jdbc-doris/src/main/java/org/apache/gravitino/catalog/doris/operation/DorisTableOperations.java

```diff
@@ -164,6 +168,37 @@ protected String generateCreateTableSql(
     return result;
   }
 
+  private Map<String, String> appendNecessaryProperties(Map<String, String> properties) {
+    Map<String, String> resultMap;
+    if (properties == null) {
+      resultMap = new HashMap<>();
+    } else {
+      resultMap = new HashMap<>(properties);
+    }
+
+    // If the backend server is less than 3, we need to set the property 'replication_allocation'
+    // to 1 explicitly.
+    if (!properties.containsKey(REPLICATION_FACTOR)) {
```

Comment:
should use `propertiesMetadata.getOrDefault` method

Reply:
I don't care about the result of `propertiesMetadata.getOrDefault` and only mind whether users have set property `REPLICATION_FACTOR` or not.

Reply:
I reviewed the code again, the method `appendNecessaryProperties` seems unnecessary since the `REPLICATION_FACTOR` already have the default value

Reply:
1. The default value will only be used when the backend server is less than 3.
2. So, if we don't explicitly set the value, how can we pass the value to the Doris cluster.

Reply:
The default value should only be used when the user does not specify the property.

Reply:
If the user sets the property value to more than 3 when the backend server is less than 3, they should provide a response with the result.

Reply:
> The default value will only be used when the backend server is less than 3.

I mean we will use the default value when the following two conditions are both true
- Users do not set the value explicitly.
- The number of backend servers is less than 3.

Reply:
Condition 2 seems unnecessary?

Reply:
How do you get this?  Please refer to more information of the related issue #3173.

Reply:
I have previously reviewed the issue. I don't understand your point; in my opinion, disregarding condition 2 is not a problem.

Reply:
Additionally, Doris has the default value of `REPLICATION_FACTOR` if the number of backend servers is greater or equal to 3. The default value for `REPLICATION_FACTOR` in Gravitino is not consistent with that in Doris.

Reply:
maybe we should use `replication_num` instead of `replication_allocation`